### PR TITLE
docs(testing): clarify testing frameworks section

### LINKED ISSUE #4897: Include installing async plugin in Testing docs
Currently if you follow the [Testing guide](https://textual.textualize.io/guide/testing/) in the docs, trying to run the tests will result in this pytest warning:

> PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.
  You need to install a suitable plugin for your async framework

It would be helpful if this guide included installing the async plugin, and perhaps also how to configure this so the async tests are marked automatically.

### PR DESCRIPTION
**Pull request by TomJGooding**
Closes #4897

### CODE REVIEW

File: docs/guide/testing.md [hunk 1]
```diff
@@ -23,9 +23,16 @@ Your test code will help you find bugs early, and alert you if you accidentally
...
+Textual is an async framework powered by Python's [asyncio](https://docs.python.org/3/library/asyncio.html) library.
+While Textual doesn't require a particular test framework, it must provide support for asyncio testing.
 
+You can use any test framework you are familiar with, but we will be using [pytest](https://docs.pytest.org/)
+along with the [pytest-asyncio](https://pytest-asyncio.readthedocs.io/) plugin in this chapter.
+
+By default, the `pytest-asyncio` plugin requires each async test to be decorated with `@pytest.mark.asyncio`.
+You can avoid having to add this marker to every async test
+by setting `asyncio_mode = auto` in your pytest configuration
+or by running pytest with the `--asyncio-mode=auto` option.
```
Comment (darrenburns) [inline docs/guide/testing.md:35]: Is this definitely accurate these days? I thought it was no longer required 🤔 I'm pretty sure I don't do this in Posting and the tests run fine.

Comment (TomJGooding) [reply docs/guide/testing.md:35]: This was my understanding from the errors trying to follow the example in the Testing guide and reading the docs.

**Comment by merriam** [comment]
Last seen this is still required but handled by the `pyproject.toml`.  Specifically in `[tool.pytest.ini_options]`, it has `asyncio_mode = "auto"`

**Comment by darrenburns** [comment]
This makes sense on reading the docs, but I'm hesitant to merge because the tests at https://github.com/darrenburns/posting run fine without it, which suggests I may be missing something.

**Review by darrenburns — APPROVED** [review]

**Comment by darrenburns** [comment]
Thanks!

**Pull request merged**